feat(codegen): expose `with_source_text` function

camc314 opened the pull request:
it is unsafe to call `print_expression` as the source text may be empty. in this case, the end user would experience panics when printing certain expressions, as it requires referencing the source text

camc314 commented:
* **#10769** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10769?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#10768** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10768?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10768?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10768).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camchenry commented on crates/oxc_codegen/src/lib.rs:
```diff
@@ -251,6 +258,8 @@ impl<'a> Codegen<'a> {
     /// Print a single [`Expression`], adding it to the code generator's
     /// internal buffer. Unlike [`Codegen::build`], this does not consume `self`.
+    ///
+    /// NOTE: you must call [`Codegen::with_source_text`] before calling this method
```
can we add a debug assertion for this?

camc314 replied:
yes. good idea, will add

camc314 committed:
feat(codegen): expose `with_source_text` function (#10768)

it is unsafe to call `print_expression` as the source text may be empty. in this case, the end user would experience panics when printing certain expressions, as it requires referencing the source text
